rdflib/plugins/sparql/operators.py

```diff
@@ -582,14 +582,62 @@ def Builtin_EXISTS(e, ctx):
     return Literal(not exists)
 
 
+_CUSTOM_FUNCTIONS = {}
+
+def register_custom_function(uri, func, override=False, raw=False):
+    """
+    Register a custom SPARQL function.
+
+    By default, the function will be passed the RDF terms in the argument list.
+    If raw is True, the function will be passed an Expression and a Context.
+
+    The function must return an RDF term, or raise a SparqlError.
+    """
+    if not override and uri in _CUSTOM_FUNCTIONS:
+        raise ValueError("A function is already registered as %s" % uri.n3())
+    _CUSTOM_FUNCTIONS[uri] = (func, raw)
+
+def custom_function(uri, override=False, raw=False):
+    """
+    Decorator version of :func:`register_custom_function`.
+    """
+    def decorator(func):
+        register_custom_function(uri, func, override=override, raw=raw)
+        return func
+    return decorator
+
+def unregister_custom_function(uri, func):
+    if _CUSTOM_FUNCTIONS.get(uri, (None, None))[0] != func:
+        raise ValueError("This function is not registered as %s" % uri.n3())
+    del _CUSTOM_FUNCTIONS[uri]
+
+    
 def Function(e, ctx):
     """
-    Custom functions (and casts!)
+    Custom functions and casts
     """
+    pair =_CUSTOM_FUNCTIONS.get(e.iri)
+    if pair is not None:
+        func, raw = pair
+        if raw:
+            return func(e, ctx)
+        else:
+            try:
+                return func(*e.expr)
+            except TypeError, ex:
```

Comment:
I can fix it, but the same syntax was used in other places, that I didn't change, so I was not sure.